From ec6c7a69747909db3ed2b5fa859cb40371e64e2d Mon Sep 17 00:00:00 2001
From: Timple Raj M <quic_timple@quicinc.com>
Date: Tue, 26 Sep 2023 15:33:59 +0530
Subject: [PATCH] samples: qmi: Resolve the warning

This patch resolves the below mentioned warning

CC [M]  samples/qmi/qmi_sample_client.o
samples/qmi/qmi_sample_client.c: In function 'test_qmi_write':
samples/qmi/qmi_sample_client.c:1372:25: warning: ignoring return
value of 'kstrtoul' declared with attribute
'warn_unused_result' [-Wunused-result]
1372 |                         kstrtoul(cmd, 0, qdentry[index].value);
     |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Change-Id: I5d3c757d1ed9a8a16304ae174a9bb5951068fd05
Signed-off-by: Timple Raj M <quic_timple@quicinc.com>
---
 samples/qmi/qmi_sample_client.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/samples/qmi/qmi_sample_client.c b/samples/qmi/qmi_sample_client.c
index d79b0f0935cf..5139d1c91001 100644
--- a/samples/qmi/qmi_sample_client.c
+++ b/samples/qmi/qmi_sample_client.c
@@ -1348,6 +1348,7 @@ static ssize_t test_qmi_write(struct file *fp, const char __user *buf,
 	unsigned char cmd[64];
 	int len;
 	int index = 0;
+	int ret;
 	struct test_qmi_data *qmi_data;
 	struct qmi_handle *qmi = fp->private_data;
 	struct qmi_sample *sample = container_of(qmi, struct qmi_sample, qmi);
@@ -1369,7 +1370,9 @@ static ssize_t test_qmi_write(struct file *fp, const char __user *buf,
 	for (index = 1; index < sizeof(qdentry)/sizeof(struct qmi_dir); index++) {
 		if (!strncmp(fp->f_path.dentry->d_iname, qdentry[index].string, \
 					sizeof(fp->f_path.dentry->d_iname))) {
-			kstrtoul(cmd, 0, qdentry[index].value);
+			ret = kstrtoul(cmd, 0, qdentry[index].value);
+			if (ret)
+				return ret;
 			return count;
 		}
 	}
-- 
2.34.1

